### DISCUSSION TITLE
Remove car from controller

### DISCUSSION THREAD

**Discussion by john-haugland** [flobz/psa_car_controller#477] [Q&A]
Hi, 
How can I remove an old car VIN from the application. The car is not visible in the iphone or android app, but it shows up in this application. I tried by removing it from cars.json and charge_config.json. Anyway, it comes back after running the psa-car-controller
psa-car-controller -f config.json -c charge_config.json -r -R 5

**Comment by nosuchuser** [comment]
I'm facing the same problem. 
So far, I've tried to contact Citroen's support (no luck, their first suggestion was to delete the vehicle from the MyCitroen app), apply for an API key on Stellantis to make the call and remove the old VIN (still no response on the request) and now I'm looking at the code to do it myself.

I'm thinking about submitting a merge request where the save_cars function checks the config file for blacklisted VINs and skips writing it to the cars.json file on initialization. Blacklisted VINs could be specified on the config file.

Any comments, @flobz  ?

**Reply by flobz** [reply]
Your solution seems good to me.

**Comment by CalicodesIles** [comment]
Hello,

Did you try to remove your car from MyCitroen website ?

I had a similar issue a while ago, and it was solved by using the
website instead of the app.

Regards.

**Comment by nosuchuser** [comment]
That's a good insight - but it doesn't work.
 
The website works so much better than the mobile app. Having said so, I gave it a go, logged in and the only car that appears is the one I want to keep. The one I deleted does not show - but the VIN is returned via API.

So far, I've managed to establish two distinct areas on the PSA car controller code where a carefully crafted "if" clause skips the execution, so the events from the other car are not processed as I originally wanted. I may look into more concise and efficient ways to do this later.

Thanks anyway!

**Reply by sehroriginell** [reply]
I have the same problem. I suspect the unwanted car to cause some trouble as I am getting 500 errors. If you know how to get rid of the car, I'd be happy for some insights. Thanks!

**Comment by gerar19** [comment]
> That's a good insight - but it doesn't work.
> 
> The website works so much better than the mobile app. Having said so, I gave it a go, logged in and the only car that appears is the one I want to keep. The one I deleted does not show - but the VIN is returned via API.
> 
> So far, I've managed to establish two distinct areas on the PSA car controller code where a carefully crafted "if" clause skips the execution, so the events from the other car are not processed as I originally wanted. I may look into more concise and efficient ways to do this later.
> 
> Thanks anyway!

Hi, please, can tell me what file and line? I have a same problem, the old vehicle always exist....

thanks

**Comment by nosuchuser** [comment]
Until Stellantis removed, at my request, the VIN from my account (which took a while), I changed get_vehicles on https://github.com/flobz/psa_car_controller/blob/master/psa_car_controller/psacc/application/psa_client.py to be the following:

```
    def get_vehicles(self):
        try:
            res = self.api().get_vehicles_by_device()
            for vehicle in res.embedded.vehicles:
                if (vehicle.vin != 'VR7BCZKXxxxxxxxxx'):
                    self.vehicles_list.add(Car(vehicle.vin, vehicle.id, vehicle.brand, vehicle.label))
            self.vehicles_list.save_cars()
        except (ApiException, InvalidHeader):
            logger.exception("get_vehicles:")
        return self.vehicles_list
```

Replace 'VR7BCZKXxxxxxxxxx' by the "bad" VIN.
It should be fairly simple to add reference data and/or configuration items to blacklist VINs
